firehose.c

```diff
@@ -29,6 +29,7 @@
  * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
  * POSSIBILITY OF SUCH DAMAGE.
  */
+#define _LARGEFILE64_SOURCE
```

Comment:
According to feature_test_macros(7) this shouldn't be used in new code, but rather _FILE_OFFSET_BITS=64 is the way to go. Do you agree?

(Also, given that off_t is signed, don't we have the problem already at 31 bits - i.e. 2GB - files?)